### PR TITLE
Listing repeating/one-off jobs and cancel functionality by id

### PR DESCRIPTION
**Pull request by farsene**

### CODE REVIEW

File: src/scheduler.py [hunk 1]
```diff
@@ -69,60 +82,98 @@ def parse_args():
...
-            at_options = '-t'
-        cmd = (f'python {os.path.abspath(args.job)} >> {output_file} '
-                f'| at {at_options} {at_time} >> /dev/null') #2>&1')
-        print(cmd)
-        subprocess.run(cmd, shell=True) # TODO: remove shell=True
-        print('scheduled one-off job')
+    isExist = os.path.exists(output_file)
+    if not isExist:
+        # Create a new directory because it does not exist
+        os.makedirs(output_file)
```
Comment (wandersiemers) [inline src/scheduler.py]: This is unnecessary: we use `>>` to write to the output, which simply creates the file if it doesn't exist

File: src/scheduler.py [hunk 2]
```diff
@@ -69,60 +82,98 @@ def parse_args():
...
+        cmd = 'atq'
+        result = check_output(cmd, universal_newlines=True)
+
+        print("One-off jobs:")
+        # results = result.strip("\n")
+        # for (idx, job) in enumerate(results):
+        #     job_info = job.strip(' ')
+        #     print(job_info)
+        #     print(f'| Id: {idx:3}|  At: {job[1]}| Queue: {job[6]:10} | User: {job[7]}|')
+        # print("Id "job number, date, hour, year, queue, and username)
```
Comment (wandersiemers) [inline src/scheduler.py]: Could you remove this code? We'll be able to get it from git, doesn't need to be in the file

**Review by wandersiemers — CHANGES_REQUESTED** [review]

File: src/scheduler.py (hunk 1 above)
Comment (farsene) [reply src/scheduler.py]: Good point, removed

File: src/scheduler.py (hunk 2 above)
Comment (farsene) [reply src/scheduler.py]: yes, removed it

**Comment by wandersiemers** [comment]
Please fix the conflicts @Farsene1

File: src/scheduler.py [hunk 3]
```diff
@@ -65,55 +76,101 @@ def parse_args():
...
+            cron.write()
+            next_run = croniter(args.repeat, datetime.datetime.now()).get_next(datetime.datetime)
+            print(f'scheduled repeating job with schedule {args.repeat} - next run at {next_run}')
+        elif args.at is not None:
+            # we have a one-off job, schedule using `at`
+            format = '%Y%m%d%H%M'  # at's format:[[CC]YY]MMDDhhmm
+            if args.green is False:
+                time_str = new_start.strftime(format)
+            else:
+                time_str = delay.strftime(format)
```
Comment (farsene) [inline src/scheduler.py]: What is up with this delay? Do we want it?

**Pull request merged**